### PR TITLE
Multi-file code fix verification

### PR DESCRIPTION
**Pull request by otac0n**
Adds support for verifying code fixes that affect more than one file.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/FileMayOnlyContainTestBase.cs [hunk 1]
```diff
@@ -32,18 +33,30 @@ public abstract class FileMayOnlyContainTestBase : CodeFixVerifier
...
 
             DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(4, this.Keyword.Length + 2);
 
             await this.VerifyCSharpDiagnosticAsync(testCode.Replace("%1", this.Keyword), expected, CancellationToken.None).ConfigureAwait(false);
-            await this.VerifyCSharpDiagnosticAsync(fixedCode.Replace("%1", this.Keyword), EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+
+            foreach (var code in fixedCode)
+            {
+                await this.VerifyCSharpDiagnosticAsync(code.Replace("%1", this.Keyword), EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            }
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/FileMayOnlyContainTestBase.cs]: :thought_balloon: This code construction is repeated a lot. Might be a good idea to add a VerifyNoCSharpDiagnosticsAsync method that will accept either a source file or a list of sources.

**Comment by vweijsters** [comment]
:+1:  Looks good to me, only one small remark.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/FileMayOnlyContainTestBase.cs (hunk 1 above)
Comment (otac0n) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/FileMayOnlyContainTestBase.cs]: There are already some methods in the CodeFixVerifier and DiagnosticVerifier that take arrays of strings. These calls treat the array as multiple files in a single project.

I feel that adding an overload that takes an array of strings, but treats each independently, would confuse the semantics of the verifiers.

Comment (otac0n) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/FileMayOnlyContainTestBase.cs]: Turns out, the overload exists.  I'll update.

**Pull request merged**